core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -595,6 +595,9 @@ private Evaluator extractInclusiveDeleteExpression(ManifestReader reader) {
           if (bin.contains(cachedNewManifest) && bin.size() < minManifestsCountToMerge) {
             // not enough to merge, add all manifest files to the output list
             outputManifests.addAll(bin);
+          } else if (bin.size() < minManifestsCountToMerge) {
```

Comment:
I think this changes the way `mergeGroup` works and might lead to never merging certain old manifests.

I would try to modify the if condition above (i.e. `bin.contains(cachedNewManifest)`) to check if the bin contains anything from appendManifests. Having said that, I am not sure about the performance.

Reply:
However, `bin` is a list, so `contains` is already O(n).

Reply:
@aokolnychyi thanks.

> I think this changes the way mergeGroup works and might lead to never merging certain old manifests.

Are we talking about this case: when old manifest files will not have the `cachedNewManifest` ( manifest for newly added files ), then with this new condition it will not merge old manifests till it reaches min.count (default = 100).  ?
But wondering, as merge will trigger only when we add new files or manifets, and manifests for a table should not merge until it reaches min.count or ? 

> However, bin is a list, so contains is already O(n).

yeah if we do check whether any of the appendManifests(m) is present in bin(n) then in worst case it would be 0(n*m).

Reply:
@aokolnychyi updated the condition.